### PR TITLE
Add option to skip material transfer for production orders

### PR DESCRIPTION
**Pull request by ishanloya**
[Forum link](https://discuss.erpnext.com/t/production-order-with-bulk-stock-transfer/21641)

### CODE REVIEW

**Comment by rmehta** [comment]
Screenshots and updated documentation please!

https://github.com/frappe/erpnext/wiki/Pull-Request-Guidelines

**Comment by ishanloya** [comment]
I'll add screenshots and documentation. Didn't understand the status bit however.

**Comment by rmehta** [comment]
If you are not tracking qty by Stock Entry, how do you complete the Production Order?

**Comment by ishanloya** [comment]
The second Stock Entry (type- Manufacture) is still there, which completes the Production Order. It's only the first entry (type- Material Transfer for Manufacture) which isn't tracked against each Production Order. This is useful if a company transfers material in bulk and cannot track the transferred material for each Production Order. Maybe it will be clearer once I add the documentation.

**Comment by ishanloya** [comment]
@rmehta Have updated the documentation. Thanks 🙂

File: erpnext/manufacturing/doctype/production_order/production_order.js [hunk 1]
```diff
@@ -235,9 +248,15 @@ $.extend(cur_frm.cscript, {
...
 	make_se: function(purpose) {
 		var me = this;
-		var max = (purpose === "Manufacture") ?
-			flt(this.frm.doc.material_transferred_for_manufacturing) - flt(this.frm.doc.produced_qty) :
-			flt(this.frm.doc.qty) - flt(this.frm.doc.material_transferred_for_manufacturing);
+		if(!this.frm.doc.skip_transfer){
+			var max = (purpose === "Manufacture") ?
+				flt(this.frm.doc.material_transferred_for_manufacturing) - flt(this.frm.doc.produced_qty) :
+				flt(this.frm.doc.qty) - flt(this.frm.doc.material_transferred_for_manufacturing);
+		} else if(this.frm.doc.skip_transfer){
```
Comment (nabinhait) [inline erpnext/manufacturing/doctype/production_order/production_order.js]: else {
     var max = flt(this.frm.doc.qty) - flt(this.frm.doc.produced_qty);
}

File: erpnext/manufacturing/doctype/production_order/production_order.js [hunk 2]
```diff
@@ -122,21 +124,32 @@ erpnext.production_order = {
...
-				if(doc.material_transferred_for_manufacturing==doc.qty) {
-					// all materials transferred for manufacturing,
-					// make this primary
+					if(doc.material_transferred_for_manufacturing==doc.qty) {
+						// all materials transferred for manufacturing,
+						// make this primary
+						btn.addClass('btn-primary');
+					}
+				}
+			} else if(frm.doc.skip_transfer){
```
Comment (nabinhait) [inline erpnext/manufacturing/doctype/production_order/production_order.js]: `else` is enough

**Review by nabinhait — CHANGES_REQUESTED** [review]

**Comment by ishanloya** [comment]
@nabinhait Thank you for reviewing

**Review by nabinhait — APPROVED** [review]

**Pull request merged**